docs/guides/aws-private-link-workspace.md

```diff
@@ -0,0 +1,246 @@
+- `databricks_account_id`: The numeric ID for your Databricks account. When you are logged in, it appears in the bottom left corner of the page.
+- `vpc_id` - The ID for the AWS VPC 
+- `network_name` - Name for your Databricks-configured network
+- `region` - AWS region
+- `existing_network_sg` - Security groups set up for the existing VPC
+- `existing_network_subnets` - Existing subnets being used for the customer managed VPC
+- `workspace_vpce_service` - Choose the region-specific service endpoint from this table.
+- `relay_vpce_service` - Choose the region-specific service from this table.
+- `vpc_cidr_block` - CIDR range for the VPC being deployed into
+- `vpce_cidr` - CIDR range for the subnet chosen for the VPC endpoint
+- `credentials_id` - Databricks workspace credential ID 
+- `storage_configuration_id` - Databricks workspace storage configuration ID
+
+This guide is provided as-is and you can use this guide as the basis for your custom Terraform module.
+
+To get started with AWS PrivateLink integration, this guide takes you throw the following high-level steps:
+- Initialize the required providers
+- Configure AWS objects
+  - A subnet dedicated to your VPC relay and workspace endpoints
+  - A security group dedicated to your VPC endpoints
+  - Two AWS VPC endpoints
+- Workspace Creation
+
+## Provider initialization
+
+Initialize [provider with `mws` alias](https://www.terraform.io/language/providers/configuration#alias-multiple-provider-configurations) to set up account-level resources. See [provider authentication](../index.md#authenticating-with-hostname,-username,-and-password) for more details.
+
+```hcl
+terraform {
+ required_providers {
+   databricks = {
+     source  = "databrickslabs/databricks"
+     version = "0.4.7"
+   }
+   aws = {
+     source = "hashicorp/aws"
+     version = "3.49.0"
+   }
+ }
+}
+
+provider "aws" {
+ region = var.region
+}
+
+// initialize provider in "MWS" mode for provisioning workspace with AWS PrivateLink
+provider "databricks" {
+ alias    = "mws"
+ host     = "https://accounts.cloud.databricks.com"
+ username = var.databricks_account_username
+ password = var.databricks_account_password
+}
+```
+
+Define the required variables
+
+```hcl
+variable databricks_account_id {}
+variable databricks_account_username {}
+variable databricks_account_password {}
+variable vpc_id {}
+variable network_name {}
+variable region {}
+variable existing_network_sg {}
+variable existing_network_subnets {}
+
+variable workspace_vpce_service {}
+variable relay_vpce_service {}
+
+variable vpc_cidr_block {}
+variable vpce_cidr {}
+
+// Use the Databricks Account API 2.0 to retrieve these two IDs below - https://docs.databricks.com/dev-tools/api/latest/account.html
+variable credentials_id {}
+variable storage_configuration_id {}
+
+locals {
+ prefix = "private-link-ws"
+}
+
+locals {
+workspace_subnet_1 = (
+ element(split(",", var.existing_network_subnets), 1)
```

Comment:
Could you elaborate more on the workspace_subnet_1, workspace_subnet_2? If the vpc's are coming form terraform can we use terraform to get the subnet ids?

Reply:
The assumption is that the VPC already exists, which is a requirement for PrivateLink setup and mention in the first paragraph. The workspace subnets have to be specified in the `databricks_mws_networks` resource in order for us to create a workspace with PrivateLink enabled. These are user-provided. Let me know if this makes sense or if there a suggestion to do something differently here.

Reply:
@rportilla-databricks @kivanc-databricks we're also using the VPC module in documentation to setup networking, because it takes less lines of code and is way more mature. Try switching to that as well

Reply:
Will make the switch, thanks.

Reply:
@nfx , for the VPC set up, the VPC module is assumed to be created (customer-managed). From the docs, it looks like the VPC module is creating the infrastructure, which is not desired for this setup. What's the requested change for the the code as it relates to the existing VPC in this PR?

Reply:
Done.